gui/src/app/FileEditor/StanFileEditor.tsx

```diff
@@ -213,5 +215,76 @@ const stringChecksum = (str: string) => {
     return hash;
 }
 
+const stancErrorsToCodeMarkers = (stancErrors: StancErrors) => {
+    const cm: CodeMarker[] = []
+    const errorsAndWarnings = [...(stancErrors.errors || []), ...(stancErrors.warnings || [])]
+    for (const x of errorsAndWarnings) {
+        if (!x) continue
+
+        // Example: Syntax error in 'main.stan', line 1, column 0 to column 1, parsing error:
+
+        let lineNumber: number | undefined = undefined
+        let startColumn: number | undefined = undefined
+        let endColumn: number | undefined = undefined
+
+        const sections = x.split(',').map(x => x.trim())
+        for (const section of sections) {
+            if ((section.startsWith('line ')) && (lineNumber === undefined)) {
+                lineNumber = parseInt(section.slice('line '.length))
+            }
+            else if ((section.startsWith('column ')) && (startColumn === undefined)) {
+                const cols = section.slice('column '.length).split(' to ')
+                startColumn = parseInt(cols[0])
+                endColumn = cols.length > 1 ? parseInt(cols[1].slice('column '.length)) : startColumn + 1
+            }
+        }
+
+        if ((lineNumber !== undefined) && (startColumn !== undefined) && (endColumn !== undefined)) {
+            const isWarning = x.toLowerCase().startsWith('warning')
```

Comment:
Thoughts on having a loop over warnings and a loop over errors to avoid needing to back this information out again? I think if the above logic to find the start/end was its own function the two loops wouldn't need to share too much duplicate code

Reply:
Done